rohmu/object_storage/azure.py

```diff
@@ -191,13 +192,15 @@ def _iter_key(self, *, path, with_metadata, deep):
                 )
 
     def delete_key(self, key):
+        plain_key = key
         key = self.format_key_for_backend(key, remove_slash_prefix=True)
         self.log.debug("Deleting key: %r", key)
         try:
             blob_client = self.conn.get_blob_client(container=self.container_name, blob=key)
             return blob_client.delete_blob()
         except azure.core.exceptions.ResourceNotFoundError as ex:  # pylint: disable=no-member
             raise FileNotFoundFromStorageError(key) from ex
+        self.notify_delete(plain_key)
```

Comment:
feels odd that these common behaviours of notification needs to be explicitly added to every provider/implementation... could this be extracted?

Reply:
how would you go about extracting it?

Reply:
FYI: This was considered an non-issue.